### DISCUSSION TITLE
A better way to export many "PointValue" to csv?

### DISCUSSION THREAD

**Discussion by zhang-rite** [idaholab/moose#17163] [Q&A General] [answered]
Hi MOOSE community

Here what I want to get from the output csv file are the time-changes of several variables at many selected points. For fewer points,  I can do this by repeatedly setting several "**PointValue**" in **Postprocessors**. However, if there are many points,  (e.g., 100),  the list becomes very long. Could anyone tell me is there a more convenient way for the purpose already designed in MOOSE?

ps: If I use **VectorPostprocessors**'s **PointValueSampler** or **LineValueSampler**, there will be many csv files and lack time information. I want one csv file with time information.

Thank you!

**Comment by aeslaughter** [comment, marked as answer]
For Vectorpostprocessors you can enable "time_data" in the CSV block to create an additional file that gives you the time information.

```
[Outputs]
  [out]
    type = CSV
    time_data = true
  []
[]
```

You can also output using the JSON format, which should give you the time information and only contain a single file. You will need to set the "vectorpostprocessors_as_reporters" option.

```
[Outputs]
  [out]
     type = JSON
    vectorpostprocessors_as_reporters = true
  []
[]
```

**Reply by aeslaughter** [reply]
I third option would be to create a custom Reporter that allows you to list the points you want to report and store those in a vector. This would be a nice contribution for MOOSE. https://mooseframework.inl.gov/syntax/Reporters/index.html
